### PR TITLE
Filter out bad locales returned by g_get_language_names

### PR DESCRIPTION
**Pull request by robert-ancell**
We're seeing issues with and "und" (undefined) locale and exceptions in applications (see https://github.com/ubuntu/app-center/issues/1659). It seems the GLib method for getting the language names doesn't clean up invalid values, so we should do that.

### CODE REVIEW

**Comment by robert-ancell** [comment]
@hbatagelo you might also be interested in reviewing.

**Review by cbracken — APPROVED** [review]
LGTM stamp from a Japanese personal seal

lgtm!

File: shell/platform/linux/fl_engine.cc [hunk 1]
```diff
@@ -150,28 +150,50 @@ static void view_removed_cb(const FlutterRemoveViewResult* result) {
...
+
+    // Ignore empty locales, caused by settings like `LANGUAGE=pt_BR:`
+    if (strcmp(locale_string, "") == 0) {
+      continue;
     }
-    if (territory != nullptr) {
-      g_ptr_array_add(locale_strings, territory);
+
+    g_autofree gchar* language = NULL;
+    g_autofree gchar* territory = NULL;
```
Comment (hbatagelo) [inline shell/platform/linux/fl_engine.cc:173]:
```suggestion
    g_autofree gchar* language = nullptr;
    g_autofree gchar* territory = nullptr;
```

**Review by hbatagelo — COMMENTED** [review]
LGTM. Just a minor nit for consistency.

**Pull request merged**